mongo/description/shared_spec_test.go

```diff
@@ -60,7 +60,9 @@ type readPref struct {
 	TagSets      []map[string]string `json:"tag_sets"`
 }
 
-func topologyKindFromString(s string) TopologyKind {
+func topologyKindFromString(t *testing.T, s string) TopologyKind {
```

Comment:
The change to accept a `testing.T` here and in `serverKindFromString` aren't strictly required, but it's good practice for us to fail on unrecognized topology/server kind strings rather than defaulting to `Unknown` as we were before, so I felt it was best to add it.

Reply:
Nice attention to detail!